src/backend/marsha/core/api/video.py

```diff
@@ -686,3 +697,77 @@ def end_sharing(
         serializer = self.get_serializer(video)
         channel_layers_utils.dispatch_video_to_groups(video)
         return Response(serializer.data)
+
+    @action(
+        methods=["patch"],
+        detail=True,
+        url_path="start-recording",
+        permission_classes=[
+            permissions.IsTokenResourceRouteObject
+            & (permissions.IsTokenInstructor | permissions.IsTokenAdmin)
+        ],
+    )
+    # pylint: disable=unused-argument
+    def start_recording(
+        self,
+        request,
+        pk=None,
+    ):
+        """Starts video recording.
+
+        Parameters
+        ----------
+        request : Type[django.http.request.HttpRequest]
+            The request on the API endpoint
+        pk: string
+            The primary key of the video
+
+        Returns
+        -------
+        Type[rest_framework.response.Response]
+            HttpResponse with the serialized video.
+        """
+        video = self.get_object()
```

Comment:
You should also check if the video is running. We should not allow recording if the live is not running